### PR TITLE
custom recipes: add comments, debug, basic e2e test

### PR DESCRIPTION
**Pull request by abeatrix**
PR includes:

- add comments and debug to Custom Recipes files
- minor cleaned up
- added a new e2e test for Custom Recipes
- Update: fixed some issues with Custom Recipes after merging main

## Test plan

Added new e2e test for Custom Recipes

### CODE REVIEW

File: lib/shared/src/chat/recipes/my-prompt.ts [hunk 1]
```diff
@@ -237,16 +239,26 @@ export class MyPrompt implements Recipe {
                 if (filesInDir.length > 0) {
                     return await populateVscodeDirContextMessage(dirUri, filesInDir.slice(0, 10))
                 }
+                const contextMessages: ContextMessage[] = []
                 const parentDirName = getParentDirName(dirPath)
                 const fileExt = currentFileName ? getFileExtension(currentFileName) : '*'
+                // Search foe files in directory with test in the name
```
Comment (philipp-spiess) [inline lib/shared/src/chat/recipes/my-prompt.ts:245]:
```suggestion
                // Search for files in directory with test in the name
```

File: lib/shared/src/chat/recipes/my-prompt.ts [hunk 2]
```diff
@@ -237,16 +239,26 @@ export class MyPrompt implements Recipe {
...
+                const testDirFiles = await vscode.workspace.findFiles(
+                    `**/${parentDirName}/test*/**/*.${fileExt}`,
+                    '**/node_modules/**',
```
Comment (philipp-spiess) [inline lib/shared/src/chat/recipes/my-prompt.ts:248]: All of this here seems heavily tie to modern JS/TS projects. Have you tested this with python or some other languages as well?

**Review by philipp-spiess — APPROVED** [review]

File: lib/shared/src/chat/recipes/my-prompt.ts (hunk 2 above)
Comment (abeatrix) [reply lib/shared/src/chat/recipes/my-prompt.ts]: good points! I have updated the code to make it more generic. Tested it in a `c` repo:
image

Comment (abeatrix) [reply lib/shared/src/chat/recipes/my-prompt.ts]: good points! I have updated the code to make it more generic. Tested it in a `c` repo:
image

**Pull request merged**